### ISSUE TITLE
Accessing external networks on 172.17.x.x from container

### ISSUE THREAD

**Issue by issacg** [docker/for-win#1838]
I am running docker-for-windows from inside a corporate network which uses the 172.16.0.0/20 space  Specifically, 172.17 and 172.19.  

My host machine is assigned an address on another subnet, but my local gateway knows how to route packets to these subnets.  I've also tried moving `bip`to `192.168.135.1/24`  to avoid conflict, but there are some other seemingly built-in networks that are conflicting.  I've found bits and pieces of advice around the network, but nothing that I've been able to put together to really understand what's happening.

One piece of advice that gave me a temporary respite was uninstalling docker for windows, creating a network adaptor on the windows host that was bound to the subnets I needed (172.19.0.0/16 for the below example), reinstalling docker-for-windows and then removing the adapter from the host machine.  

This worked for a day or so, although shamefully, I admit it was a classic example of not really understanding what I was doing, and it eventually just stopped working.

### Expected behavior
I should be able to access external machines on these subnets from inside a container, for example `docker run --rm alpine ping 172.19.9.192`

### Actual behavior
Packets disappear, seemingly never making it to the host machine

### Information
Although running ifconfig off the default docker network doesn't shed any light on the issue, presumably, these are network interfaces being used by the MobyLinuxVM.  When inspecting the MobyLinuxVM's network stack, I can see bridges bound to these subnets

`docker run --rm -ti --privileged --network=none --pid=host justincormack/nsenter1`

```
/ # ifconfig
br-35e713676a4f Link encap:Ethernet  HWaddr 02:42:73:93:75:C8
          inet addr:172.19.0.1  Bcast:172.19.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

br-551aef06cdcb Link encap:Ethernet  HWaddr 02:42:58:A1:8B:B9
          inet addr:172.18.0.1  Bcast:172.18.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

br-65d8eec756eb Link encap:Ethernet  HWaddr 02:42:D3:F8:38:D2
          inet addr:172.21.0.1  Bcast:172.21.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

br-6ce5c3cdd6da Link encap:Ethernet  HWaddr 02:42:03:48:DE:20
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

br-775bdb469cc4 Link encap:Ethernet  HWaddr 02:42:32:E0:4B:8F
          inet addr:172.20.0.1  Bcast:172.20.255.255  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

docker0   Link encap:Ethernet  HWaddr 02:42:FA:9F:33:B9
          inet addr:192.168.135.1  Bcast:192.168.135.255  Mask:255.255.255.0
          inet6 addr: fe80::42:faff:fe9f:33b9/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:369 errors:0 dropped:0 overruns:0 frame:0
          TX packets:31 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:30240 (29.5 KiB)  TX bytes:2266 (2.2 KiB)

eth0      Link encap:Ethernet  HWaddr 02:50:00:00:00:01
          inet addr:192.168.65.3  Bcast:192.168.65.255  Mask:255.255.255.0
          inet6 addr: fe80::214a:1aa9:a869:d292/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:55 errors:0 dropped:0 overruns:0 frame:0
          TX packets:59 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:24667 (24.0 KiB)  TX bytes:7972 (7.7 KiB)

hvint0    Link encap:Ethernet  HWaddr 00:15:5D:63:01:0B
          inet addr:10.0.75.2  Bcast:0.0.0.0  Mask:255.255.255.0
          inet6 addr: fe80::215:5dff:fe63:10b/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1835 errors:0 dropped:2 overruns:0 frame:0
          TX packets:204 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:211137 (206.1 KiB)  TX bytes:19976 (19.5 KiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:38 errors:0 dropped:0 overruns:0 frame:0
          TX packets:38 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:3760 (3.6 KiB)  TX bytes:3760 (3.6 KiB)

veth1111583 Link encap:Ethernet  HWaddr 2A:E3:2D:DB:5A:3E
          inet6 addr: fe80::28e3:2dff:fedb:5a3e/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:133 errors:0 dropped:0 overruns:0 frame:0
          TX packets:17 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:12810 (12.5 KiB)  TX bytes:1206 (1.1 KiB)
```

Am I correct in assuming that this is what's causing my routing "black hole"?  Is there any workaround to coerce these bridges to change their networking config?  And, if someone can be bothered to explain it, what are these bridges for, anyway?

Thanks!

**Comment by issacg** [comment]
gentle ping

**Comment by issacg** [comment]
I stumbled on the solution.  I noticed that the "trigger" for everything breaking was the creation of some named networks as part of a compose file I'm working with, and that the above bridges were the gateways for said networks.

Further digging finds that I can control the networking of named networks in compose from https://docs.docker.com/compose/compose-file/#ipam

Once I assigned specific subnets to these networks, the problem went away.
